darts/utils/data/timeseries_dataset.py

```diff
@@ -0,0 +1,118 @@
+                              If it is set, the provided sequence must have the same length as that of `data_series`.
+        :param lh: A `(min_lh, max_lh)` interval for the forecast point, starting from the end of the series.
+                   For example, `(1, 3)` will select forecast points uniformly between `1*H` and `3*H` points
+                   before the end of the series.
+                   If `min_lh < 1`, the "targets" in the emitted (data, target) splits might be shorter than
+                   the horizon `H`. Use this only with models that are capable of masking appropriately the series
+                   during training.
+        :param lookback: A `(min_lb, max_lb)` interval for the length of the "data" in the emitted (data, target) splits.
+                         For instance, `(2, 7)` will emit "data" of lengths between `2H` and `7H`, depending
+                         on the maximum size allowed by the series' length and the selection of the forecast point.
+        """
+
+        self.data_series = [data_series] if isinstance(data_series, TimeSeries) else data_series
+        if target_series is None:
+            self.target_series = data_series
+        else:
+            self.target_series = [target_series] if isinstance(target_series, TimeSeries) else target_series
+        self.H = horizon
+        self.min_lh, self.max_lh = lh
+        self.min_lb, self.max_lb = lookback
+
+        # Checks
+        raise_if_not(self.min_lh <= self.max_lh,
+                     'The lh parameter should be an int tuple (min_lh, max_lh), with min_lh <= max_lh')
+        raise_if_not(self.min_lb <= self.max_lh,
+                     'The lookback parameter should be an int tuple (min_lb, max_lb), with min_lb <= max_lb')
+        raise_if_not(len(self.data_series) == len(self.target_series),
+                     'The provided sequence of target series must have the same length as '
+                     'the provided sequence of data series.')
+
+        self.nr_samples_per_ts = (self.max_lh - self.min_lh)*self.H * (self.max_lb - self.min_lb)
+        self.ideal_nr_samples = len(self.data_series) * self.nr_samples_per_ts
+
+    def __len__(self):
+        """
+        Returns the (ideal) number of possible (data, target) splits.
+        The actual number of possible splits might be lower, if some time series are shorter than
+         `(max_lb + max_lh) * H`.
+        """
+        return self.ideal_nr_samples
+
+    def __getitem__(self, idx):
+        ts_idx = idx // self.nr_samples_per_ts  # index of the time series
+        lh_lb_idx = idx - (ts_idx * len(self.data_series))  # index of the (data, target) split within the series
+
+        lh_idx = lh_lb_idx // (self.max_lb - self.min_lb)  # index of the actual l_h, determining the forecasting point
+        lb_idx = lh_lb_idx - (lh_idx * (self.max_lh - self.min_lh)*self.H)  # index of which lookback period to use
```

Comment:
I'm not sure that I understand exactly what is computed here and how, could you explain ?